api/agent/call.go

```diff
@@ -53,26 +55,41 @@ const (
 	invokePath = "/invoke"
 )
 
+var skipTriggerHeaders = map[string]bool{
+	"Connection":        true,
+	"Keep-Alive":        true,
+	"Trailer":           true,
+	"Transfer-Encoding": true,
+	"TE":                true,
+	"Upgrade":           true,
+}
+
 // Sets up a call from an http trigger request
 func FromHTTPTriggerRequest(app *models.App, fn *models.Fn, trigger *models.Trigger, req *http.Request) CallOpt {
 	return func(c *call) error {
-		ctx := req.Context()
-
-		log := common.Logger(ctx)
-		// Check whether this is a CloudEvent, if coming in via HTTP router (only way currently), then we'll look for a special header
-		// Content-Type header: https://github.com/cloudevents/spec/blob/master/http-transport-binding.md#32-structured-content-mode
-		// Expected Content-Type for a CloudEvent: application/cloudevents+json; charset=UTF-8
 		contentType := req.Header.Get("Content-Type")
-		t, _, err := mime.ParseMediaType(contentType)
-		if err != nil && contentType != "" {
-			// won't fail here, but log
-			log.Debugf("Could not parse Content-Type header: %v %v", contentType, err)
-		} else {
-			if t == ceMimeType {
-				c.IsCloudEvent = true
-				fn.Format = models.FormatCloudEvent
+		// transpose trigger headers into HTTP
+		headers := make(http.Header)
+		for k, vs := range req.Header {
+			// should be generally unnecessary but to be doubly sure.
+			k = textproto.CanonicalMIMEHeaderKey(k)
+			if skipTriggerHeaders[k] {
+				continue
 			}
+			rewriteKey := fmt.Sprintf("Fn-Http-H-%s", k)
+			for _, v := range vs {
+				headers.Add(rewriteKey, v)
+			}
+		}
+		requestUrl := reqURL(req)
+
+		headers.Set("Fn-Http-Method", req.Method)
+		if contentType != "" {
+			headers.Set("Content-Type", contentType)
```

Comment:
think we need to skip Content-Type above, as well, or we'll have `Fn-Http-H-Content-Type` and `Content-Type` (or was this a goal?)... not a deal breaker, anyway